### DISCUSSION TITLE
Add "On click" action to URL component

### DISCUSSION THREAD

**Discussion by deliqs** [Budibase/budibase#9268] [Ideas]
Hi,

I've searched through the discussions, but I have not found any similar feature requests. 

I have a dashboard with a table. Each row has a URL component. I'd like to add an "On click" action to this URL component. In my specific case, I'd like to keep track of what has been seen.

If there is a similar feature request, then please point me to that and delete this one.

**Comment by melohagan** [comment]
Hey @deliqs 

In my opinion, it's not an ideal design pattern to execute an action upon clicking a hyperlink. It should really just be a navigation link. 

If you want to perform actions on click, you could use a button - there is the **Quiet** setting that removes the border.
Alternatively you could put some text inside a [Container](https://docs.budibase.com/docs/container#on-click) which support on click actions.

**Comment by deliqs** [comment]
@melohagan 

That is understandable. Do you know how I can open a link in another tab via the Button component?

**Reply by melohagan** [reply]
Ah I see, no at the moment you can't do that. What is the flow of your use-case, why do you need to both fire an action and a new tab to be opened?

**Reply by deliqs** [reply]
I want to keep track of what pages have been opened. I currently have a URL component next to a checkbox. The users can click the checkbox if the page has been opened, but I want to automate that (fill the checkbox) when they click the URL component.

**Reply by melohagan** [reply]
Is that to verify users have read some terms and conditions? I'm wondering if you could maybe pull in data from the webpage into your app instead and display it in a modal or something like that?

Depends on what the links are for and why you want to keep track of the opened links.

**Reply by deliqs** [reply]
No, the table contains data which is used by sales agents. The data is scraped from multiple websites. So the sales agents want to know which pages have been viewed so that they do not spend double time on a single object.

**Reply by melohagan** [reply]
I've raised an feature request here to allow the **Navigate To** action to open links in a new tab: https://github.com/Budibase/budibase/issues/9286

**Reply by deliqs** [reply]
Perfect, thanks a lot @melohagan!
